File: core/evidence/evidence_sufficiency_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def assess_evidence_sufficiency(evidence: List[str], required: int = 2) -> Dict[str, Any]:
    count = len(evidence or [])
    sufficient = count >= required
    return {
        "sufficient": sufficient,
        "evidence_count": count,
        "required": required,
        "status": "sufficient" if sufficient else "insufficient_evidence",
        "deterministic_inputs": [f"count={count}", f"required={required}"],
    }

File: src/evidence/evidenceSufficiencyEngine.ts
/**
 * Converted from Python: core/evidence/evidence_sufficiency_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function assessEvidenceSufficiency(evidence: any, required: any = 2): any {
  var count: any = py.len(py.or2(evidence, () => ([])));
  var sufficient: any = py.ge(count, required);
  return {"sufficient": sufficient, "evidence_count": count, "required": required, "status": (py.truthy(sufficient) ? "sufficient" : "insufficient_evidence"), "deterministic_inputs": [`count=${py.toStr(count)}`, `required=${py.toStr(required)}`]};
}
